### PR TITLE
QL 1.21: Finalize change notes for JavaScript

### PR DESCRIPTION
**Pull request by felicity-semmle**
This PR tidies up the JavaScript change notes ready for the release of 1.21.

@Semmle/js

### CODE REVIEW

File: change-notes/1.21/analysis-javascript.md [hunk 1]
```diff
@@ -28,26 +28,27 @@
...
 | Incomplete regular expression for hostnames | More results | This rule now tracks regular expressions for host names further. |
 | Incomplete string escaping or encoding | More results | This rule now considers the flow of regular expressions literals, and it no longer flags the removal of trailing newlines. |
+| Incorrect suffix check | Fewer false-positive results | This rule now recognizes valid checks in more cases. |
 | Password in configuration file | Fewer false positive results | This query now excludes passwords that are inserted into the configuration file using a templating mechanism or read from environment variables. Results are no longer shown on LGTM by default. |
 | Replacement of a substring with itself | More results | This rule now considers the flow of regular expressions literals. |
 | Server-side URL redirect       | Fewer false-positive results | This rule now treats URLs as safe in more cases where the hostname cannot be tampered with. |
+| Tainted path | More results and fewer false-positive results | This rule now analyses path manipulation code more precisely. |
```
Comment (xiemaisi) [inline change-notes/1.21/analysis-javascript.md]:
```suggestion
| Tainted path | More results and fewer false-positive results | This rule now analyzes path manipulation code more precisely. |
```

Comment (felicity-semmle) [reply change-notes/1.21/analysis-javascript.md]: Thanks

**Comment by felicity-semmle** [comment]
Ah - just realized that I raised this against the wrong branch 👎. What's the most efficient way to get these changes onto the `rc/1.21` branch?

**Comment by xiemaisi** [comment]
Oh, I hadn't noticed that. Just raise a PR with the same commits against `rc/1.21`, that way we shouldn't get any conflicts.

**Comment by felicity-semmle** [comment]
Okay - thanks.

**Pull request merged**